Makefile.build

```diff
@@ -630,15 +627,73 @@ test-suite: world byte $(ALLSTDLIB).v
 # Default rules for compiling ML code
 ###########################################################################
 
-# Target for libraries .cma and .cmxa
+gramlib__pack:
+	mkdir -p $@
+
+# gramlib.ml contents
+gramlib__pack/gramlib.ml: | gramlib__pack
+	echo " \
+module Ploc    = Gramlib__Ploc \
+module Plexing = Gramlib__Plexing \
+module Gramext = Gramlib__Gramext \
+module Grammar = Gramlib__Grammar" > $@
+
+gramlib__pack/gramlib__Ploc.mli: gramlib/ploc.mli | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Ploc.ml: gramlib/ploc.ml | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Plexing.mli: gramlib/plexing.mli | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Plexing.ml: gramlib/plexing.ml | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Gramext.mli: gramlib/gramext.mli | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Gramext.ml: gramlib/gramext.ml | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Grammar.mli: gramlib/grammar.mli | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+gramlib__pack/gramlib__Grammar.ml: gramlib/grammar.ml | gramlib__pack
+	cp -a $< $@
+	sed -e "1i # 1 \"$<\"" -i $@
+
+# Specific rules for gramlib to pack it Dune / OCaml 4.08 style
+_G=gramlib__
+GRAMOBJS=$(addsuffix .cmo, $(GRAMFILES))
+
+gramlib__pack/gramlib.cmo: gramlib__pack/gramlib.ml
+	$(SHOW)'OCAMLC    $<'
+	$(HIDE)$(OCAMLC) $(COND_BYTEFLAGS) -w -49 -no-alias-deps -o $@ -c $<
 
-# The dependency over the .mllib is somewhat artificial, since
-# ocamlc -a won't use this file, hence the $(filter-out ...) below.
-# But this ensures that the .cm(x)a is rebuilt when needed,
-# (especially when removing a module in the .mllib).
-# We used to have a "order-only" dependency over .mllib.d here,
-# but the -include mechanism should already ensure that we have
-# up-to-date dependencies.
+gramlib__pack/gramlib.cmx: gramlib__pack/gramlib.ml
+	$(SHOW)'OCAMLOPT  $<'
+	$(HIDE)$(OCAMLOPT) $(COND_OPTFLAGS) $(HACKMLI) -w -49 -no-alias-deps -o $@ -c $<
+
+gramlib__pack/gramlib__%.cmi: gramlib__pack/$(_G)%.mli
```

Comment:
I don't get why these rules use the abbreviation `$(_G)` on the right but not on the left.
I thik you should drop the definition of `_G` altogether (preferred solution), or use it everywhere (here and in the rules above that I've commented).